packages/synapse-react-client/src/components/SqlDefinedTableEditor/SqlDefinedTableEditorModal.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react'
+import ConfirmationDialog from '../ConfirmationDialog'
+import SqlDefinedTableEditor from './SqlDefinedTableEditor'
+import { useGetEntity, useUpdateEntity } from '../../synapse-queries'
+import { convertToEntityType } from '../../utils/functions/EntityTypeUtils'
+import { MaterializedView, VirtualTable } from '@sage-bionetworks/synapse-types'
+import { Alert } from '@mui/material'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+export type SqlDefinedTableEditorModalProps = {
+  open: boolean
+  entityId: string
+  onUpdate: () => void
+  onCancel: () => void
+}
+export default function SqlDefinedTableEditorModal(
+  props: SqlDefinedTableEditorModalProps,
+) {
+  const { open, entityId, onCancel, onUpdate } = props
+  const {
+    data: entity,
+    isLoading: isEntityLoading,
+    error: entityError,
+  } = useGetEntity<MaterializedView | VirtualTable>(entityId)
+  const [sql, setSql] = useState('')
+
+  useEffect(() => {
+    if (entity) {
+      setSql(entity.definingSQL)
+    }
+  }, [entity])
+
+  const type = entity ? convertToEntityType(entity.concreteType) : undefined
+  const {
+    mutate,
+    isLoading: isUpdateLoading,
+    error: updateError,
+  } = useUpdateEntity<MaterializedView | VirtualTable>({ onSuccess: onUpdate })
+
+  return (
+    <ConfirmationDialog
+      open={open}
+      title="Update SQL"
+      content={
+        <>
+          <SqlDefinedTableEditor
+            value={sql}
+            entityType={type}
+            onChange={e => setSql(e.target.value)}
+            disabled={isEntityLoading || isUpdateLoading}
+          ></SqlDefinedTableEditor>
+          {(entityError || updateError) && (
+            <Alert severity="error">
+              {entityError?.reason || updateError?.reason}
+            </Alert>
+          )}
```

Comment:
Corresponding with the above change:
```suggestion
          {error && (
            <Alert severity="error">
              {error.reason}
            </Alert>
          )}
```

This is more straightforward and lets us get rid of the optional chaining